rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/500_date_range.yml

```diff
@@ -0,0 +1,119 @@
+setup:
+
+  - do:
+      indices.create:
+        index: dates_year_only
+        body:
+          mappings:
+            properties:
+              date:
+                type: date
+                format: uuuu
+
+  - do:
+      bulk:
+        refresh: true
+        body:
+          - '{ "index" : { "_index" : "dates_year_only", "_id" : "first" } }'
+          - '{"date" : "1900", "field" : 1 }'
+          - '{ "index" : { "_index" : "dates_year_only", "_id" : "second" } }'
+          - '{"date" : "2022", "field" : 1 }'
+          - '{ "index" : { "_index" : "dates_year_only", "_id" : "third" } }'
+          - '{"date" : "2022", "field" : 2 }'
+          - '{ "index" : { "_index" : "dates_year_only", "_id" : "fourth" } }'
+          - '{"date" : "1500", "field" : 2 }'
+
+  - do:
+      indices.create:
+        index: dates
+        body:
+          mappings:
+            properties:
+              date:
+                type: date
+
+  - do:
+      bulk:
+        refresh: true
+        body:
+          - '{ "index" : { "_index" : "dates", "_id" : "first" } }'
+          - '{"date" : "1900-01-01T12:12:12.123456789Z", "field" : 1 }'
+          - '{ "index" : { "_index" : "dates", "_id" : "second" } }'
+          - '{"date" : "2022-01-01T12:12:12.123456789Z", "field" : 1 }'
+          - '{ "index" : { "_index" : "dates", "_id" : "third" } }'
+          - '{"date" : "2022-01-03T12:12:12.123456789Z", "field" : 2 }'
+          - '{ "index" : { "_index" : "dates", "_id" : "fourth" } }'
+          - '{"date" : "1500-01-01T12:12:12.123456789Z", "field" : 2 }'
+          - '{ "index" : { "_index" : "dates", "_id" : "fifth" } }'
+          - '{"date" : "1500-01-05T12:12:12.123456789Z", "field" : 2 }'
+
+---
+"test range query for all docs with year uuuu":
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: dates
+        body:
+          query:
+            range:
+              date:
+                gte: 1000
+                lte: 2023
+                format: uuuu
+
+  - match: { hits.total: 5 }
+  - length: { hits.hits: 5 }
+
+---
+"test match query gte and lt for single result with year uuuu":
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: dates
+        body:
+          query:
+            range:
+              date:
+                gte: 1500 #1500-01-01T00:00:00
+                lte: 1500 #1500-01-01T23:59:59
```

Comment:
but did we had that behaviour in the past? It might be a bug on its own, but I thought that this is how it always worked

Reply:
Testing on a 7x install it seems you're right, this has always been broken.  But it's still a bug.  Is it simple to fix here or should we merge this PR as-is and work on that separately?

Reply:
agree this is a bug (or a missing feature at least). I think this would require some more work. It would make 2 cases for rounding (up and down) I am happy to raise an issue where we could gather all the use cases